pkg/cloud/services/compute/instance.go

```diff
@@ -505,6 +508,28 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 	}
 
 	record.Eventf(eventObject, "SuccessfulCreatePort", "Created port %s with id %s", port.Name, port.ID)
+
+	if portOpts.DisablePortSecurity != nil {
+		mc2 := metrics.NewMetricPrometheusContext("port", "update")
+		portUpdateOpts := ports.UpdateOpts{}
+
+		if *portOpts.DisablePortSecurity {
```

Comment:
The same here. 
if the type is just bool, it will be false by default. 
Basically, 513 and 514 are not needed unless the value is true and ca be moved to after 516

Reply:
There's a critical difference here. There are genuinely 3 behaviours:

* DisablePortSecurity = true -> Disable port security
* DisablePortSecurity = false -> Do not disable port security
* DisablePortSecurity = nil -> Inherit port security setting from the network

It's detailed here: https://docs.openstack.org/api-ref/network/v2/?expanded=create-port-detail#id53

Reply:
Thanks for clarifying it. One case though, What happens for the following case. 
NetworkLevel = disable && PortLevel=enable ?

Reply:
it should always follow what is set at the port level